azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -563,25 +582,14 @@ func resourceArmKubernetesClusterCreateUpdate(d *schema.ResourceData, meta inter
 	kubernetesVersion := d.Get("kubernetes_version").(string)
 
 	linuxProfile := expandKubernetesClusterLinuxProfile(d)
+	windowsProfile := expandKubernetesClusterWindowsProfile(d)
 	agentProfiles := expandKubernetesClusterAgentPoolProfiles(d)
 	servicePrincipalProfile := expandAzureRmKubernetesClusterServicePrincipal(d)
 	networkProfile := expandKubernetesClusterNetworkProfile(d)
 	addonProfiles := expandKubernetesClusterAddonProfiles(d)
 
 	tags := d.Get("tags").(map[string]interface{})
 
-	// we can't do this in the CustomizeDiff since the interpolations aren't evaluated at that point
-	if networkProfile != nil {
-		// ensure there's a Subnet ID attached
-		if networkProfile.NetworkPlugin == containerservice.Azure {
-			for _, profile := range agentProfiles {
-				if profile.VnetSubnetID == nil {
-					return fmt.Errorf("A `vnet_subnet_id` must be specified when the `network_plugin` is set to `azure`.")
-				}
-			}
-		}
-	}
```

Comment:
can we add this logic back in?

Reply:
Not Really:
* https://github.com/Azure/acs-engine/issues/1767
* https://github.com/Azure/acs-engine/pull/2168
That's why I deleted it

Reply:
Correct, AKS with windows required Azure CNI, and now, the AzureCNI doesn't require a custom VNET / Subnet to be provided.

Reply:
based on this comment: https://github.com/Azure/acs-engine/pull/2168/files#diff-f45af0b14c127459636440a5981a6934R823

> custom VNET not yet supported

this is still valid for linux clusters, and is planned to come to Windows, so perhaps we need to add this back in with an `if !windows` check?

Reply:
It appears this wasn't added back in?

Reply:
@katbyte @tombuildsstuff Are you sure you want it bacK ? The test was removed one year ago: https://github.com/Azure/acs-engine/pull/2322

Reply:
@tombuildsstuff was mentioning it is still required for linux clusters and that linked issue only refers to windows.

Reply:
the customVnet is supported, we are currently using it with hybrid cluster.

looks at the latest code:

https://github.com/jackfrancis/acs-engine/blob/master/pkg/api/vlabs/validate.go#L609-L651

Reply:
https://github.com/Azure/acs-engine/pull/2322 this when it was reverted. which shows that customVnet is supported with Windows